homeassistant/components/discovergy/config_flow.py

```diff
@@ -57,35 +52,20 @@ class DiscovergyConfigFlow(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
-    _existing_entry: ConfigEntry
-
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> ConfigFlowResult:
-        """Handle the initial step."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user",
-                data_schema=CONFIG_SCHEMA,
-            )
-
-        return await self._validate_and_save(user_input)
-
     async def async_step_reauth(
         self, entry_data: Mapping[str, Any]
     ) -> ConfigFlowResult:
         """Handle the initial step."""
-        self._existing_entry = self._get_reauth_entry()
         return await self.async_step_reauth_confirm()
```

Comment:
You can drop `async_step_reauth_confirm` altogether
```suggestion
        return await self.async_step_user()
```